### PR TITLE
`config.md`: Changes to section about TOML

### PR DESCRIPTION
**Pull request by ilyagr**
Based on @martinvonz 's suggestion from https://github.com/martinvonz/jj/pull/1263.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -20,11 +20,11 @@ name = "YOUR NAME"
 For a more complicated example,
 
 ```toml
-[template-aliases]
-"format_short_id(id)" = "id.shortest(12)"
+[colors]
+"commit_id prefix" = { bold = true}
```
Comment (samueltardieu) [inline docs/config.md:24]: I'd write "For more complicated examples" and let both examples, as the `template-aliases` example illustrates defining a function. Both are useful to me.

Comment (ilyagr) [reply docs/config.md:24]: Interesting idea. I'm wondering if this means I should rearrange things a little, perhaps extract this to a separate subsection.

Comment (ilyagr) [reply docs/config.md]: I added both examples and rewrote things a little.

File: docs/config.md [hunk 2]
```diff
@@ -39,12 +48,10 @@ easier to write down in an unconfusing way. If you are confident with TOML
 then use whichever suits you in your config. If you mix dotted keys and headings,
 **put the dotted keys before the first heading**.
 
-The other thing to remember is that the value of a setting (the part to the
-right of the `=` sign) should be surrounded in quotes if it's a string. That's
+That's
 probably enough TOML to keep you out of trouble but the [syntax guide] is very
 short if you ever need to check.
```
Comment (martinvonz) [inline docs/config.md:53]: nit: The new text looks weirdly wrapped with just "That's" on a single line. Could you rewrap it?

**Review by martinvonz — APPROVED** [review]

**Pull request merged**